pyogrio/_io.pyx

```diff
@@ -196,28 +205,33 @@ cdef void* ogr_open(const char* path_c, int mode, char** options) except NULL:
     else:
         flags |= GDAL_OF_READONLY
 
-
     try:
         # WARNING: GDAL logs warnings about invalid open options to stderr
         # instead of raising an error
-        ogr_dataset = exc_wrap_pointer(
-            GDALOpenEx(path_c, flags, NULL, <const char *const *>options, NULL)
-        )
-
-        return ogr_dataset
+        with capture_errors() as errors:
+            ogr_dataset = GDALOpenEx(path_c, flags, NULL, <const char *const *>options, NULL)
+            return errors.check_pointer(ogr_dataset)
 
     except NullPointerError:
         raise DataSourceError(
-            "Failed to open dataset (mode={}): {}".format(mode, path_c.decode("utf-8"))
+            f"Failed to open dataset (mode={mode}): {path_c.decode('utf-8')}"
         ) from None
 
     except CPLE_BaseError as exc:
-        if str(exc).endswith("a supported file format."):
+        # If there are inner exceptions, append their errmsg to the errmsg
+        errmsg = str(exc)
+        inner = exc.__cause__
+        while inner is not None:
+            errmsg = f"{errmsg}; {inner}"
+            inner = inner.__cause__
```

Comment:
Is this not something we want to do in the logic in `_err.pyx` ?

Reply:
I've been wondering about this as well.

I'll add a parameter "squash_errors" or something like to the `check_...` functions that to have the choice in having the errors returned as a stack of exceptions or as one exception with all error messages "squashed" to one.

Reply:
Done.